target/linux/ramips/dts/mt7621_mikrotik_routerboard-7xx.dtsi

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		mode {
+			label = "mode";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	beeper {
+		compatible = "gpio-beeper";
+		gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		usb_power {
+			gpio-export,name = "usb_power";
+			gpio-export,output = <1>;
+			gpios = <&gpio 12 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <20000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "RouterBoot";
+				reg = <0x0 0x40000>;
+				read-only;
+				compatible = "mikrotik,routerboot-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partition@0 {
+					label = "bootloader1";
+					reg = <0x0 0x0>;
+					read-only;
+				};
+
+				hard_config: hard_config {
+					read-only;
+				};
+
+				partition@10000 {
+					label = "bootloader2";
+					reg = <0x10000 0xf000>;
+					read-only;
+				};
+
+				soft_config {
+				};
```

Comment:
Hi @f00b4r0, I could not manage to set ethonce in softconfig for routerboot with rbsysfs soft_config.
Is this expected with what is setup here?

This is trivial, but I wanted to check I haven't missed something.
```
root@OpenWrt:~# ls /sys/firmware/mikrotik/soft_config/
bios_version   boot_device    boot_proto     cpu_mode       silent_boot
boot_delay     boot_key       booter         cpufreq_index  uart_speed
root@OpenWrt:~# cat /sys/firmware/mikrotik/soft_config/boot_device 
eth [flasheth] ethonce flash cfg cfgonce
root@OpenWrt:~# echo "ethonce" > /sys/firmware/mikrotik/soft_config/boot_device 
-ash: can't create /sys/firmware/mikrotik/soft_config/boot_device: Permission denied
root@OpenWrt:~# echo "2" > /sys/firmware/mikrotik/soft_config/boot_device 
-ash: can't create /sys/firmware/mikrotik/soft_config/boot_device: Permission denied
```

Cheers

Reply:
`ls -l /sys/firmware/mikrotik/soft_config/` ?
I think this is not a bug. Most likely 4K_SECTORS isn't enabled on this target.

Reply:
Yes. I skimmed straight over that in the commit message, sorry.
```
grep 4K_SECTORS build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.4.45/.config
# CONFIG_MTD_SPI_NOR_USE_4K_SECTORS is not set
CONFIG_MTD_SPI_NOR_USE_4K_SECTORS_LIMIT=4096
root@OpenWrt:~# ls -l /sys/firmware/mikrotik/soft_config/
-r--------    1 root     root          4096 Jun 13 12:36 bios_version
-r--------    1 root     root          4096 Jun 13 12:36 boot_delay
-r--------    1 root     root          4096 Jun 13 12:36 boot_device
-r--------    1 root     root          4096 Jun 13 12:36 boot_key
-r--------    1 root     root          4096 Jun 13 12:36 boot_proto
-r--------    1 root     root          4096 Jun 13 12:36 booter
-r--------    1 root     root          4096 Jun 13 12:36 cpu_mode
-r--------    1 root     root          4096 Jun 13 12:36 cpufreq_index
-r--------    1 root     root          4096 Jun 13 12:36 silent_boot
-r--------    1 root     root          4096 Jun 13 12:36 uart_speed
root@OpenWrt:~# cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00040000 00010000 "RouterBoot"
mtd1: 0000f000 00010000 "bootloader1"
mtd2: 00001000 00010000 "hard_config"
mtd3: 0000f000 00010000 "bootloader2"
mtd4: 00001000 00010000 "soft_config"
mtd5: 00001000 00010000 "bios"
mtd6: 00fc0000 00010000 "firmware"
mtd7: 00250000 00010000 "kernel"
mtd8: 00d70000 00010000 "rootfs"
mtd9: 00b30000 00010000 "rootfs_data"
```

Reply:
So operation is exactly as designed: read-only access. Not a bug 😄